File: 1260.py
n, m, v = map(int, input().split())
connect = [ ([False] * n) for i in range(n) ]

for i in range(m):
	a, b = map(int, input().split())
	a -= 1
	b -= 1
	connect[a][b] = connect[b][a] = True

def dfs (point, opened):
	global n, m, connect
	print( point + 1, end=" " )
	opened[ point ] = True
	for i in range(n):
		if i == point:
			continue
		if connect[point][i] and not opened[ i ]:
			dfs(i, opened)

def bfs (point):
	global n, m, connect
	stack = [point]
	opened = [ False ] * n
	opened[point] = True
	while stack:
		item = stack.pop(0)
		print( item + 1, end=" " )
		for i in range(n):
			if i == item:
				continue
			if connect[item][i] and not opened[ i ]:
				stack += [i]
				opened[ i ] = True

dfs( v-1, [ False ] * n )
print()
bfs( v-1 )

File: 1260.swift
var arr = readLine()!.split(separator: " ")
var n = Int(arr[0])!
var m = Int(arr[1])!
var v = Int(arr[2])! - 1

var connect : [[Bool]] = [[Bool]](repeating: [Bool](repeating: false, count: n), count: n)

for i in 0..<m {
    arr = readLine()!.split(separator: " ")
    let from = Int(arr[0])! - 1
    let to = Int(arr[1])! - 1
    connect[from][to] = true
    connect[to][from] = true
}

var opened : [Bool] = [Bool](repeating: false, count: n)
func dfs ( point : Int ) {
    opened[point] = true
    print(point+1, terminator: " ")
    for i in 0..<n {
        if i == point { continue }
        if connect[point][i] && !opened[i] { dfs(point: i) }
    }
}
dfs(point: v)
print()

opened = [Bool](repeating: false, count: n)
func bfs ( point : Int ) {
    opened[point] = true
    var queue : [Int] = [Int]()
    queue.append( point )
    while queue.count > 0 {
        var poped = queue[0]
        queue.remove(at: 0)
        print(poped + 1, terminator: " ")
        for i in 0..<n {
            if i == poped { continue }
            if connect[poped][i] && !opened[i] {
                opened[i] = true
                queue.append( i )
            }
        }
    }
}
bfs(point: v)
